File: backend/src/main/java/com/ap/campaign_manager/campaign/Campaign.java
package com.ap.campaign_manager.campaign;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.util.*;

@Entity
@NoArgsConstructor
@Getter
@Setter
public class Campaign {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private String campaignName;

    @ElementCollection
    @CollectionTable(name = "campaign_keywords", joinColumns = @JoinColumn(name = "campaign_id"))
    @Column(nullable = false, name = "keyword")
    private Set<String> keywords = new HashSet<>();

    @Column(nullable = false, precision = 12, scale = 2)
    private BigDecimal bidAmount;

    @Column(nullable = false, precision = 12, scale = 2)
    private BigDecimal campaignFund;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private CampaignStatus status;

    @Column(nullable = false)
    private String town;

    @Column(nullable = false)
    private Integer radius;
}

File: frontend/src/types/campaign.ts
export type CampaignStatus = 'ON' | 'OFF'

export interface Campaign {
    id: string
    campaignName: string
    keywords: string[]
    bidAmount: string
    campaignFund: string
    status: CampaignStatus
    town: string
    radius: number
}

export interface CampaignRequest {
    campaignName: string
    keywords: string[]
    bidAmount: string
    campaignFund: string
    status: CampaignStatus
    town: string
    radius: number
}